TAScheduler/models.py

```diff
@@ -3,21 +3,31 @@
 from typing import Optional
 
 
+
 class UserType(models.TextChoices):
     ADMIN = "A", "Administrator"
     PROF = "P", "Professor"
     TA = "T", "TA"
 
     @classmethod
-    def from_str(cls, maybe_type: str) -> UserType:
-        if maybe_type == 'A':
-            return UserType.ADMIN
-        elif maybe_type == 'P':
-            return UserType.PROF
-        elif maybe_type == 'T':
-            return UserType.TA
-        else:
-            raise TypeError(f'user_type {maybe_type} is non in the set of [A, P, T]')
+    def from_str(cls, maybe_type: str, user_types: dict = None) -> UserType:
+        if user_types is None:
```

Comment:
Should probably Have documentation
```suggestion
        def from_str(cls, maybe_type: str, user_types: dict = None) -> UserType
            """
            Try to translate a UserType string, as stored in the database and returned from templates, into a
            UserType object. Must be in the set ["A", "P", "T"].
            
            Raises TypeError if it is not in said set.
            """
            if user_types is None:
```